roachtest: use RepeatRunE in dmsetupDiskStaller

tbg opened the pull request:
We seem to have an issue across multiple branches (at least 24.2 and 23.1) where the dmesetup-based disk staller occasionally hiccups with

> device-mapper: reload ioctl on data1 (253:0) failed: Device or resource busy

and the test fails.

This PR adds a band-aid around the corresponding invocation in the hopes
that a retry will address the issue. If nothing else, the reference to
an issue next to the occasionally failing invocation will help
coordinate further efforts should they be required.

See
https://github.com/cockroachdb/cockroach/issues/129619#issuecomment-2309628187.

Closes #126550.
Closes #128761.
Closes #129047.
Closes #129306.
Closes #129619.

Release note: None
Epic: None

tbg committed:
roachtestutil: add RepeatRunE

This is cribbed from the `roachtest/tests` package. For ease of backporting, the
original repeatRunE remains and no callers are updated (yet).

Release note: None
Epic: None

tbg committed:
roachtestutil: augment Fataler

This preps it for supporting RepeatRunE, which wants
to stop trying when the test has failed.

Release note: None
Epic: None

tbg committed:
roachtest: use RepeatRunE in dmsetupDiskStaller

We seem to have an issue across multiple branches (at least 24.2 and 23.1) where the dmesetup-based disk staller occasionally hiccups with

> device-mapper: reload ioctl on data1 (253:0) failed: Device or resource busy

and the test fails.

This PR adds a band-aid around the corresponding invocation in the hopes
that a retry will address the issue. If nothing else, the reference to
an issue next to the occasionally failing invocation will help
coordinate further efforts should they be required.

See
https://github.com/cockroachdb/cockroach/issues/129619#issuecomment-2309628187.

Release note: None
Epic: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/129626)
<!-- Reviewable:end -->

tbg committed:
roachtest: turn repeatRunE into a passthrough

This avoids code divergence in the future. Really repeatRunE should
simply be removed, but this would churn across multiple roachtests and
thus should be done separately.

Epic: none
Release note: None

tbg committed:
roachtest: remove (unused) unused lint suppression

`RepeatRunE` has callers now.

Epic: none
Release note: None

tbg commented:
Ran a smoke check, `--- PASS: failover/non-system/disk-stall (1227.26s)` so I didn't obviously break anything here.

herkolategan reviewed (commented):
`RunE` on https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/cluster/cluster_interface.go#L117 (`Cluster interface`) now takes an `install.RunOptions` which also has retry capabilities (see https://github.com/cockroachdb/cockroach/blob/master/pkg/roachprod/install/run_options.go#L72). Probably out of scope for this PR, but it should be able to simplify or deprecate `repeateRunE`.

edit: Although this may not backport easily to 23.1

tbg commented:
Thanks for the review - I'm actually going to abandon this PR since it turned out there is a better fix for the issue I'm looking at: https://github.com/cockroachdb/cockroach/issues/129619#issuecomment-2316147244

It seems worth to boil the ocean at some point and remove these various hand-crafted legacy retry loops, but there are a number of them and they're used all over the place - a great way to get sidetracked and cause noise.

I'll clean up any further ones I come across though.
